CRM/Financial/BAO/FinancialItem.php

```diff
@@ -260,16 +263,19 @@ public static function checkContactPresent($contactIds, &$error) {
    * @return array
    */
   public static function getPreviousFinancialItem($entityId) {
-    $params = [
-      'entity_id' => $entityId,
-      'entity_table' => 'civicrm_line_item',
-      'options' => ['limit' => 1, 'sort' => 'id DESC'],
-    ];
-    $salesTaxFinancialAccounts = civicrm_api3('FinancialAccount', 'get', ['is_tax' => 1]);
-    if ($salesTaxFinancialAccounts['count']) {
-      $params['financial_account_id'] = ['NOT IN' => array_keys($salesTaxFinancialAccounts['values'])];
+    $financialItemAPI = FinancialItem::get(FALSE)
+      ->addWhere('entity_id', '=', $entityId)
+      ->addWhere('entity_table', '=', 'civicrm_line_item')
+      ->addOrderBy('id', 'DESC');
+
+    $salesTaxFinancialAccounts = FinancialAccount::get(FALSE)
+      ->addSelect('id')
+      ->addWhere('is_tax', '=', 1)
+      ->execute();
+    if ($salesTaxFinancialAccounts->count() > 0) {
+      $financialItemAPI->addWhere('financial_account_id', 'NOT IN', $salesTaxFinancialAccounts->column('id'));
     }
-    return civicrm_api3('FinancialItem', 'getsingle', $params);
+    return $financialItemAPI->execute()->first();
```

Comment:
If we want to keep this identical the equivalent of 'getsingle' is 
```suggestion
    return $financialItemAPI->execute()->single();
```

Reply:
@colemanw I didn't know `->single()` existed! But I don't think we need it in this instance. The previous code was passing limit=1 and then id=DESC which forces a single result. Then it was being retrieve with getsingle(). A fairly common code pattern since it was a bit of a pain getting the first result from a standard api3 get() (ie. you had to use reset() or similar).

Reply:
> I didn't know `->single()` existed!

Yea Tim added it for parity with api3.getsingle. It's just like `->first()` but throws an exception if there isn't exactly 1 result.